presto-testng-services/src/main/java/io/prestosql/testng/services/ReportUnannotatedMethods.java

```diff
@@ -77,17 +76,33 @@ private static boolean isTestMethod(Method method)
         if (method.getDeclaringClass() == Object.class) {
             return true;
         }
-        Class<?> superclass = method.getDeclaringClass().getSuperclass();
-        Method overridden;
+        if (method.getDeclaringClass().isInterface()) {
+            return isTemptoClass(method.getDeclaringClass());
+        }
+        for (Class<?> interfaceClass : method.getDeclaringClass().getInterfaces()) {
+            if (isTemptoClass(interfaceClass)) {
+                return true;
+            }
+            Optional<Method> overridden = getOverridden(method, interfaceClass);
+            if (overridden.isPresent()) {
+                return true;
+            }
+        }
+
+        return getOverridden(method, method.getDeclaringClass().getSuperclass())
+                .map(ReportUnannotatedMethods::isAllowedPublicMethodInTest)
+                .orElse(false);
+    }
+
+    private static Optional<Method> getOverridden(Method method, Class<?> base)
+    {
         try {
             // Simplistic override detection
-            overridden = superclass.getMethod(method.getName(), method.getParameterTypes());
+            return Optional.of(base.getMethod(method.getName(), method.getParameterTypes()));
         }
         catch (NoSuchMethodException ignored) {
-            return false;
+            return Optional.empty();
         }
-
-        return isTestMethod(overridden);
```

Comment:
You apparently removed recursion from here.

eg what if A.test overrides B.test and B.test overrides C.test and C.test is `@Test` annotated?